### PR TITLE
feat(connector): [Payme] Support Apple Pay 

### PR DESCRIPTION
**Pull request by Sangamesh26**
## Type of Change

- [x] New feature

## Description

Support Apple Pay for PayMe

Steps followed:
1. Session call: We will get an empty object with delayed session and connector as `payme` and next-action as `confirm`
2. confirm call: we hit `generate-sale` endpoint of PayMe to get payme-seller-id
3. Hyperswitch provides next_action that sdk has to use whcih contains `connector_reference_id`
4. Payment will reach to terminal status in sdk, so, there should be a retrieve call to hyperswitch that updates the status in the system

### Additional Changes

- [x] This PR modifies the API contract
- [x] This PR modifies application configuration/environment variables

Added `payme` in `connectors_with_delayed_session_response`, as payme also follows delayed session flow

## Motivation and Context

Add support for Apple Pay through PayMe

## How did you test it?

Local testing not possible

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code

### CODE REVIEW

File: crates/router/src/connector/payme/transformers.rs [hunk 1]
```diff
@@ -232,8 +263,8 @@ impl TryFrom<&PaymentMethodData> for SalePaymentMethod {
     fn try_from(item: &PaymentMethodData) -> Result<Self, Self::Error> {
         match item {
             PaymentMethodData::Card(_) => Ok(Self::CreditCard),
-            PaymentMethodData::Wallet(_)
-            | PaymentMethodData::PayLater(_)
+            PaymentMethodData::Wallet(_) => Ok(Self::ApplePay), //Later when Google Pay comes, we can have a check for wallets
```
Comment (ArjunKarthik) [inline crates/router/src/connector/payme/transformers.rs:266]: Let's have condition for ApplePay

File: crates/router/src/connector/payme/transformers.rs [hunk 2]
```diff
@@ -287,6 +318,67 @@ impl TryFrom<&types::RefundSyncRouterData> for PaymeQueryTransactionRequest {
...
+                                amount: amount_in_base_unit,
+                            },
+                            merchant_capabilities: None,
+                            supported_networks: None,
+                            merchant_identifier: None,
+                        }),
+                        connector: "payme".to_string(),
+                        delayed_session_token: true,
+                        sdk_next_action: api_models::payments::SdkNextAction {
+                            next_action: api_models::payments::NextActionCall::InvokeSdkAndEnd,
```
Comment (ArjunKarthik) [inline crates/router/src/connector/payme/transformers.rs:370]: add sync also

**Review by ArjunKarthik — CHANGES_REQUESTED** [review]
Please take changes required for cards also in this PR

**Review by ArjunKarthik — CHANGES_REQUESTED** [review]

**Comment by ArjunKarthik** [comment]
@Sangamesh26 If this PR is not required, can you close this?

**Comment by Sangamesh26** [comment]
Closing this, as same has been covered in #1922

**Pull request closed without merging**